internal/controller/supervisorconfig/githubupstreamwatcher/github_upstream_watcher.go

```diff
@@ -47,40 +60,49 @@ type UpstreamGitHubIdentityProviderICache interface {
 }
 
 type gitHubWatcherController struct {
+	namespace                      string
 	cache                          UpstreamGitHubIdentityProviderICache
 	log                            plog.Logger
 	client                         supervisorclientset.Interface
 	gitHubIdentityProviderInformer idpinformers.GitHubIdentityProviderInformer
 	secretInformer                 corev1informers.SecretInformer
+	httpClientBuilder              func(rootCAs *x509.CertPool) *http.Client
 }
 
 // New instantiates a new controllerlib.Controller which will populate the provided UpstreamGitHubIdentityProviderICache.
 func New(
+	namespace string,
 	idpCache UpstreamGitHubIdentityProviderICache,
 	client supervisorclientset.Interface,
 	gitHubIdentityProviderInformer idpinformers.GitHubIdentityProviderInformer,
 	secretInformer corev1informers.SecretInformer,
 	log plog.Logger,
 	withInformer pinnipedcontroller.WithInformerOptionFunc,
+	httpClientBuilder func(rootCAs *x509.CertPool) *http.Client,
 ) controllerlib.Controller {
 	c := gitHubWatcherController{
+		namespace:                      namespace,
 		cache:                          idpCache,
 		client:                         client,
 		log:                            log.WithName(controllerName),
 		gitHubIdentityProviderInformer: gitHubIdentityProviderInformer,
 		secretInformer:                 secretInformer,
+		httpClientBuilder:              httpClientBuilder,
 	}
 
 	return controllerlib.New(
 		controllerlib.Config{Name: controllerName, Syncer: &c},
 		withInformer(
 			gitHubIdentityProviderInformer,
-			pinnipedcontroller.MatchAnythingFilter(pinnipedcontroller.SingletonQueue()),
+			pinnipedcontroller.SimpleFilter(func(obj metav1.Object) bool {
```

Comment:
Should we add this to each of the other controllers?  Should they all filter by namespace?  (Not here, but we may want to do a follow PR to the other controllers to add this).